app/assets/javascripts/listings.js

```diff
@@ -2,17 +2,57 @@ var loadData = function(){
   $.ajax({
     type: 'GET',
     url: '/visualization.json',
-    async: false,
     dataType: 'json',
     success: function(data){
+      var listings = data.filter(function(point){
+        return point.latitude && point.discriminatory;
+      });
       drawPieChart(data);
-      console.log('success');
+      initMap(listings);
     },
     failure: function(result){
       alert('ERROR');
     }
   })
 }
+
+function initMap(listings) {
+  var map = new google.maps.Map(document.getElementById('map'), {
+    zoom: 11,
+    center: {lat: 42.3601, lng: -71.0589}
+  });
+  var labels = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+  if (listings) {
+    var markers = listings.map(function(location, i) {
+      var temp = {lat: parseFloat(location.latitude), lng: parseFloat(location.longitude)}
+      return new google.maps.Marker({
+        position: temp,
+        label: labels[i % labels.length]
+      });
+    });
+    var geoJson = [];
+    for (var listing in listings) {
+      var temp = {
+        type: 'Feature',
+        geometry: {type: 'Point', coordinates: [parseInt(listing.latitude, 10), parseInt(listing.longitude, 10)]},
```

Comment:
Remove `parseInt(listing.latitude, 10), parseInt(listing.longitude, 10)` after parsing JSON on a successful response and just have `[listing.latitude, listing.longitude]`